### PR TITLE
Added docs for delete and search posts for apiv4

### PR DESCRIPTION
**Pull request by ruzette**

### CODE REVIEW

File: source/v4/posts.yaml [hunk 1]
```diff
@@ -70,6 +70,31 @@
           $ref: '#/responses/Unauthorized'
         '403':
           $ref: '#/responses/Forbidden'
+    delete:
+      tags:
+        - posts
+      summary: Deletes a post
```
Comment (jwilander) [inline source/v4/posts.yaml]: Minor but can we make this `Delete a post`?

File: source/v4/posts.yaml [hunk 2]
```diff
@@ -70,6 +70,31 @@
...
+      description: |
+        Deletes a single post.
+        ##### Permissions
+        Must be logged in as the user or have `manage_system` permission.
```
Comment (jwilander) [inline source/v4/posts.yaml]: I think it's actually "logged in as the user who created the post or have the `delete_others_posts` permission." right?

File: source/v4/posts.yaml [hunk 3]
```diff
@@ -70,6 +70,31 @@
...
+      parameters:
+        - name: post_id
+          in: path
+          description: ID of the post to get
```
Comment (jwilander) [inline source/v4/posts.yaml]: Should be "ID of the post to delete"

File: source/v4/posts.yaml [hunk 4]
```diff
@@ -70,6 +70,31 @@
           $ref: '#/responses/Unauthorized'
         '403':
           $ref: '#/responses/Forbidden'
+    delete:
+      tags:
+        - posts
+      summary: Deletes a post
+      description: |
+        Deletes a single post.
```
Comment (jwilander) [inline source/v4/posts.yaml]: Can you update this to be "Soft deletes a post, by marking the post as deleted in the database. Soft deleted posts will not be returned in post queries."? Later we'll add endpoints for permanently deleting posts

**Comment by jwilander** [comment]
@ruzette can you give this a rebase when you get a chance?

**Comment by ruzette** [comment]
@jwilander rebased and updated the descriptions per feedback. Ready for re-review. :)

**Pull request merged**